### PR TITLE
Add Onboarding step for Interests

### LINKED ISSUE #14: [Onboarding] Interest Selection Page (FE-2)
Create the interests selection page for hobbies/preferences and gift restrictions

**Hobbies**:
Image

**Gift Restrictions**
Image

### Acceptance Criteria
- [ ] Build multi-select interface:
  - [ ] Min 3, max 10 categories
  - [ ] Pre-populated category options
- [ ] Add hobbies free-text input
- [ ] Add "gifts to avoid" field
- [ ] Update progress bar!
- [ ] Add validation messages

### PR DESCRIPTION
**Pull request by kepsteen**
Closes #14 

- created a multi-select input component
  - uses hidden checkbox and styled labels
- added step to the onboarding form with hobby category selection and interests text area
 
 
Recording 2024-12-10 at 15 28 11

### CODE REVIEW

File: components/ui/textarea.tsx
Comment (alexappleget) [inline components/ui/textarea.tsx:1]: Not: just to keep things organized and easy to find. Can you replace the "ui" folder name as "TextArea"? That way we know what file is in it

File: app/onboarding/page.tsx [hunk 1]
```diff
@@ -2,247 +2,370 @@
...
-  SelectItem,
-  SelectTrigger,
-  SelectValue,
+	Select,
+	SelectContent,
+	SelectItem,
+	SelectTrigger,
+	SelectValue,
 } from "@/components/Select/select";
+import { MultiSelect } from "@/components/Multi-Select/multi-select-input";
```
Comment (alexappleget) [inline app/onboarding/page.tsx:35]: This is most likely the cause of your build failure. It has "Multi-Select" but the folder is actually "Multi-select" lowercase the "s"

File: package-lock.json [hunk 2]
```diff
@@ -1340,6 +1341,39 @@
         }
       }
     },
+    "node_modules/@radix-ui/react-slider": {
```
Comment (alexappleget) [inline package-lock.json:1344]: Are you using another ui for these? Other than shadcn. Just curious about all the new package.json changes

**Review by alexappleget — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 2 above)
Comment (guardiansofthegit-collab) [reply package-lock.json:1344]: I created a new branch off of this one to start working on the last onboarding step where I used the slider, so I think that got included in this PR somehow.

**Review by alexappleget — APPROVED** [review]

**Pull request merged**